finagle-base-http/src/main/scala/com/twitter/finagle/http/TraceInfo.scala

```diff
@@ -81,7 +86,9 @@ private object TraceInfo {
         request.headerMap.add(Header.Sampled, sampled.toString)
       case None => ()
     }
-    request.headerMap.add(Header.Flags, JLong.toString(traceId.flags.toLong))
+    if (traceId.flags.toLong != 0L) {
```

Comment:
ps I added this check because in the file and noticed we were burning some headers for no benefit.